### PR TITLE
Remove old snapshotStorage and replication

### LINKED ISSUE #1050: SnapshotReplicationTest fails on Windows
`io.zeebe.broker.it.clustering.SnapshotReplicationTest` fails 100% of the time.

A test case fails with something like the following stacktrace, and then all remaining tests fail with address-already-in-use errors.

```
java.lang.RuntimeException: java.nio.file.FileSystemException: C:\Users\LINDHA~1\AppData\Local\Temp\279db8e3-7a65-4120-beba-056e5ed110d5\data\internal-system-0\snapshots\snap-1.snapshot: Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird.

	at io.zeebe.test.util.AutoCloseableRule.after(AutoCloseableRule.java:44)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:50)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:538)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:760)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:460)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:206)
Caused by: java.nio.file.FileSystemException: C:\Users\LINDHA~1\AppData\Local\Temp\279db8e3-7a65-4120-beba-056e5ed110d5\data\internal-system-0\snapshots\snap-1.snapshot: Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird.

	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:86)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:269)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at io.zeebe.util.FileUtil$1.visitFile(FileUtil.java:52)
	at io.zeebe.util.FileUtil$1.visitFile(FileUtil.java:1)
	at java.nio.file.Files.walkFileTree(Files.java:2670)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at io.zeebe.util.FileUtil.deleteFolder(FileUtil.java:46)
	at io.zeebe.broker.it.clustering.ClusteringRule.lambda$0(ClusteringRule.java:100)
	at io.zeebe.test.util.AutoCloseableRule.after(AutoCloseableRule.java:42)
	... 17 more
```

### LINKED ISSUE #1105: Remove snapshot replication
Once stream processors run in read-only mode on follower partitions, snapshot replication will become obsolete, as each follower partition will already have its own state/snapshots.

Blocked by #1104

### LINKED ISSUE #1731: Remove snapshortStorage usage from stream processors
Since all processors are migrated to RocksDB they all use now the StateControllers/SnapshotControllers, we can remove the SnapshotStorages from the builders and maybe refactor a bit the StreamProcessController.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
* remove snapshotStorage usage from StreamProcessors
 * remove Snapshot replication
 * remove some more unused classes

closes #1731 
closes #1050
closes #1105

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/logstreams/processor/StreamProcessorIds.java [hunk 1]
```diff
@@ -17,32 +17,6 @@
...
-
-  public static final int DISTRIBUTE_PROCESSOR_ID = 60;
-
-  public static final int WORKFLOW_INSTANCE_PROCESSOR_ID = 70;
-
-  public static final int INCIDENT_PROCESSOR_ID = 80;
-
-  public static final int MESSAGE_PROCESSOR_ID = 90;
-
   public static final int EXPORTER_PROCESSOR_ID = 1003;
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/logstreams/processor/StreamProcessorIds.java]: If this is the only constant in this file, let's move it to the exporter stream processor, no?

File: broker-core/src/main/resources/management-schema.xml [hunk 2]
```diff
@@ -39,31 +39,6 @@
...
-    <field name="chunkOffset" id="2" type="uint64"/>
-    <field name="chunkLength" id="3" type="int32"/>
-    <data name="name" id="4" type="varDataEncoding"/>
-  </sbe:message>
-
-  <sbe:message name="FetchSnapshotChunkResponse" id="9">
-    <data name="data" id="0" type="blob"/>
-  </sbe:message>
-
   <sbe:message name="ErrorResponse" id="10">
```
Comment (npepinpe) [inline broker-core/src/main/resources/management-schema.xml]: You removed the `ErrorResponse` class, so we should probably remove this from the protocol as well. Or is it used somewhere else?

File: logstreams/src/test/java/io/zeebe/logstreams/processor/FunctionProcessor.java [hunk 3]
```diff
@@ -16,7 +16,6 @@
 package io.zeebe.logstreams.processor;
 
 import io.zeebe.logstreams.log.LoggedEvent;
-import io.zeebe.logstreams.spi.SnapshotSupport;
 import java.util.function.Consumer;
 
 public class FunctionProcessor implements StreamProcessor, EventProcessor {
```
Comment (npepinpe) [inline logstreams/src/test/java/io/zeebe/logstreams/processor/FunctionProcessor.java]: Do we actually use this?

**Review by npepinpe — COMMENTED** [review]
+100 −4,225 :tada: 

Do we still need the old snapshot controller abstraction layer if everything uses RocksDB snapshots? I mean that would fit in another issue, but is there any issue for it?

File: broker-core/src/main/java/io/zeebe/broker/logstreams/processor/StreamProcessorIds.java (hunk 1 above)
Comment (Zelldon) [reply broker-core/src/main/java/io/zeebe/broker/logstreams/processor/StreamProcessorIds.java]: aggree

File: broker-core/src/main/resources/management-schema.xml (hunk 2 above)
Comment (Zelldon) [reply broker-core/src/main/resources/management-schema.xml]: I was not sure :man_shrugging: I think it is still used with ErrorResponseWriter and on some other places

File: logstreams/src/test/java/io/zeebe/logstreams/processor/FunctionProcessor.java (hunk 3 above)
Comment (Zelldon) [reply logstreams/src/test/java/io/zeebe/logstreams/processor/FunctionProcessor.java]: Yes in some tests :sweat_smile:

**Comment by Zelldon** [comment]
> Do we still need the old snapshot controller abstraction layer if everything uses RocksDB snapshots? I mean that would fit in another issue, but is there any issue for it?

LogBlockIndex uses still the old Snapshotting stuff #1730

**Comment by Zelldon** [comment]
@npepinpe I removed the FS impl and some other unused stuff. I would like not to remove the interface in this issue since I will rework this also in #1195

**Review by npepinpe — APPROVED** [review]

**Pull request merged**